cores/arduino/pins_arduino.h

```diff
@@ -60,108 +60,161 @@ enum {
 
 // Analog pins must be contiguous to be able to loop on each value
 #define MAX_ANALOG_INPUTS 24
+
 _Static_assert(NUM_ANALOG_INPUTS <= MAX_ANALOG_INPUTS,
                "Core NUM_ANALOG_INPUTS limited to MAX_ANALOG_INPUTS");
+#ifndef PIN_A0
 _Static_assert(NUM_ANALOG_FIRST >= NUM_ANALOG_INPUTS,
                "First analog pin value (A0) must be greater than or equal to NUM_ANALOG_INPUTS");
+#endif
 
 // Defined for backward compatibility with Firmata which unfortunately use it
+#ifndef AEND
 #define AEND (NUM_ANALOG_FIRST+NUM_ANALOG_INPUTS)
+#endif
 
 #if NUM_ANALOG_INPUTS > 0
+#ifndef PIN_A0
```

Comment:
This could be add on the previous line and avoid adding an `#endif`

Same for all following `#ifndef PIN_Ax`

Reply:
Do you mean like
```c
#if !defined(PIN_A0) && NUM_ANALOG_INPUTS > 0
```

Reply:
Yes but I'm wrong. I've just figure out that 
`static const uint8_t A0 = PIN_A0;`
was made here.
So, this is correct, my bad.